reprounzip-singularity/Singularity_multiple_runs.py

```diff
@@ -0,0 +1,135 @@
+# Singularity unpacker for reprozip 
+import subprocess
+import os
+import sys
+import re
+from shutil import copyfile
+import yaml
+
+#TODO:
+# 1. Logging error for every bash command
+# 2. Using existing reprozip utility functions for common tasks like copying busy box
+# 3. Comments
+# 4. integrating with exisitng unpacking coding design
+# 5. Improving code structure
+
+def extract_reprozip_file(filename):
+    bashCommand = " tar -xf {}".format(filename)
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+    data_file="DATA.tar.gz"
+    if data_file:
+        bashCommand = " tar -xf {}".format(data_file)
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
```

Comment:
You can use `--strip-components=1` to get rid of the `DATA` name. But using [`tarfile`](https://docs.python.org/3/library/tarfile.html) is the way you should go eventually, so that you don't have to write `DATA.tar.gz` to disk, and because eventually you are writing into a new tar file instead of a disk folder.